packages/builder/src/components/backend/DataTable/buttons/grid/GridDevProdSwitcher.svelte

```diff
@@ -17,23 +17,18 @@
   $: tableDatasource = $datasources.list.find(datasource => {
     return datasource._id === $tables.selected?.sourceId
   })
-  $: disabled = !isDeployed && !tableDatasource?.usesEnvironmentVariables
+  $: hasProductionData = isDeployed || tableDatasource?.usesEnvironmentVariables
```

Comment:
Why does `tableDatasource?.usesEnvironmentVariables` imply that there is production data?